std/experimental/allocator/building_blocks/kernighan_ritchie.d

```diff
@@ -172,6 +172,7 @@ struct KRRegion(ParentAllocator = NullAllocator)
     private void[] payload;
     private Node* root;
     private bool regionMode = true;
+    private size_t bytesUsedRegionMode = 0;
```

Comment:
Interesting approach. Could you please eliminate `regionMode`? Make it distinguished by `bytesUsedRegionMode == size_t.max`. So then:

```d
private bool regionMode() const { return bytesUsedRegionMode != size_t.max; }
private void cancelRegionMode() { bytesUsedRegionMode = size_t.max; }
```